### ISSUE TITLE
remove legacy drive conventions and functions

### ISSUE THREAD

**Issue by rythorpe** [jonescompneurolab/hnn-core#393]
I think this merits it's own issue. Quoting @cjayb,

OK, `drives.py:drive_event_times` is the remains of `feeds.py`, i.e., deprecated (used to be called `feed_event_times`). The test `test_drives.py:test_external_drive_times` is basically testing that the expected "default" drives are read from the params-file, and that the gid-assignments are 'correct' insofar as they reproduce default GUI output.

That can all go, though I suppose we should immediately open a follow-up PR to implement tests of `drives.py:_drive_cell_event_times`!

I'd also like to point out that I think this hack might be worth removing in this PR

https://github.com/jonescompneurolab/hnn-core/blob/d6b08ff60d9a74d517582755d5e0d6d4dbd3df89/hnn_core/network.py#L589-L595

The tricky part is here for cases when all weights are zero (at least is used to be a problem before this PR...):

https://github.com/jonescompneurolab/hnn-core/blob/d6b08ff60d9a74d517582755d5e0d6d4dbd3df89/hnn_core/network_builder.py#L474-L477

_Originally posted by @cjayb in https://github.com/jonescompneurolab/hnn-core/issues/383#issuecomment-874841427_

**Comment by rythorpe** [comment]
> OK, `drives.py:drive_event_times` is the remains of `feeds.py`, i.e., deprecated (used to be called `feed_event_times`). The test `test_drives.py:test_external_drive_times` is basically testing that the expected "default" drives are read from the params-file, and that the gid-assignments are 'correct' insofar as they reproduce default GUI output.
> 
> That can all go, though I suppose we should immediately open a follow-up PR to implement tests of `drives.py:_drive_cell_event_times`!

I totally agree, let's get rid of `drives.drive_event_times()` in favor of building tests off of `drives._drive_cell_event_times`.

> I'd also like to point out that I think this hack might be worth removing in this PR
> 
> https://github.com/jonescompneurolab/hnn-core/blob/d6b08ff60d9a74d517582755d5e0d6d4dbd3df89/hnn_core/network.py#L589-L595

I'm actually in favor of making
```python
target_populations = list(self.cell_types.keys())
```
standard practice (not just legacy) because it ensures that artificial drive cells are created irrespective of the location of the drive or specific cell types within the network. This seems like good practice as it simplifies the logic for how/when an artificial cell gets created.

**Comment by cjayb** [comment]
Hmm, interesting! I think this makes a lot of sense after your work in  #383 Either way, removes the need for the legacy-flag

**Comment by jasmainak** [comment]
@rythorpe does something remain unaddressed in this PR after #369 ?

**Comment by rythorpe** [comment]
Nope.

**Comment by rythorpe** [comment]
Any residual issue(s) were addressed in #431.
